In [1]:
%run '../lib/libraries.ipynb'

dython              0.6.1
tinydb              4.2.0


In [2]:
dfRaw = load_df('dfRaw')

##### Investigate binary variables assignment_attempt_number and assignment_max_attempts
Appears binary but the variable name indicates it could have any values. The data only contains 1 and 0


In [3]:
def num_max_cross():
    dfPd = dfRaw.select('assignment_attempt_number', 'assignment_max_attempts').toPandas()
    # Return cross-tabulation table of assignment_attempt_number vs assignment_max_attempts adding counts for null values
    return pd.crosstab(dfPd.assignment_attempt_number.fillna('null'), dfPd.assignment_max_attempts.fillna('null'), margins=True, margins_name="Total")

num_max_cross()

assignment_max_attempts       0      1  Total
assignment_attempt_number                    
0                          1566      0   1566
1                             0  78982  78982
Total                      1566  78982  80548

- both variables only contain "0" 1566 out of 78,982 observations, otherwise the value is always "1"
- are they the same 1566 as the "[unassigned]" in response_correctness

###### Verify same variables as the  1566 as the "[unassigned]" in response_correctness

In [4]:
dfRaw.where( ( F.col('response_correctness') == '[unassigned]') & (F.col('assignment_max_attempts') == 0) & (F.col('assignment_attempt_number') == 0)).select().count()

1566

- This is related to the 1566 as the "[unassigned]" in response_correctness
- Address in the investigation of the 1566

In [5]:
finish_todo("Investigate binary variables assignment_attempt_number and assignment_max_attempts")